common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java

```diff
@@ -249,6 +249,86 @@ public void testEndsWith() throws SparkException {
     assertEndsWith("äbćδe", "ÄBcΔÉ", "UNICODE_CI", false);
   }
 
+  private void assertStringInstr(String string, String substring, String collationName,
```

Comment:
this is important: Vladimir noticed that we've been using assertEquals incorrectly in `CollationSupportSuite.java`

please fix this according to his PR changes: https://github.com/apache/spark/pull/46058

since all of the asserts pass here, we wouldn't notice this, but the correct order is: first **expected** value, and then **received** value